omegaconf/basecontainer.py

```diff
@@ -435,7 +436,11 @@ def assign(value_key: Any, value_to_assign: Any) -> None:
             elif input_config:
                 assign(key, value)
             else:
-                self.__dict__["_content"][key] = wrap(key, value)
+                if change_node:
```

Comment:
can you use the should_set_value for this instead of introducing another variable that changes the control flow of this complex function?

Try to extend it to cover this case too.